packages/nx/src/adapter/ngcli-adapter.ts

```diff
@@ -123,7 +120,14 @@ export async function scheduleTarget(
       const { json: packageJson, path: packageJsonPath } =
         readPluginPackageJson(
           nodeModule,
-          this.workspaces['resolvePaths'].bind(this.workspaces)()
+          this.root
+            ? [
+                this.root,
+                __dirname,
+                process.cwd(),
+                ...getNxRequirePaths(this.root),
+              ]
+            : [__dirname, process.cwd(), ...getNxRequirePaths()]
```

Comment:
What's going on here? This doesn't seem like the same as the `resolvePaths` method in `Workspaces`.

`Workspaces` was originally instantiated with `this.root` here.

So the logic should be the following `this.root ? [this.root, __dirname] : [__dirname]` right?

I mean `__dirname` is not exactly the same... but i'm not sure if it matters :thinking: 

Whether we resolve from `packages/nx/src/config` (`__dirname` of `Workspaces`) or `packages/nx/src/adapter` (`__dirname`, of `ngcli-adapter`), I don't think makes much of a difference.

Reply:
how about `this.root ? getNxRequirePaths(this.root) : getNxRequirePaths()`? if not set, readPluginPackageJson will use getNxRequirePaths as default is not passed in